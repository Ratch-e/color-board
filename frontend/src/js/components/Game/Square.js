import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Square = ({color, click, x, y}) => (
	<div
		className={classNames('board__square', `board__square_color_${color}`)}
		onClick={() => click(x, y, color)}
	/>
);

Square.propTypes = {
	color: PropTypes.string,
	click: PropTypes.func,
	x: PropTypes.number,
	y: PropTypes.number,
};

export default Square;