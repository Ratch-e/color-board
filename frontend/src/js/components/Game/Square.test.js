import React from 'react';
import {shallow} from 'enzyme';
import Square from './Square';

describe('Квадрат на поле', () => {
	it('рендерится корректно', () => {
		const component = shallow(<Square/>);
		expect(component.length).toMatchSnapshot();
	});

	it('выбирает нужный класс в зависимости от переданного цвета', () => {
		const component = shallow(<Square color="red"/>);
		expect(component.find('.board__square').hasClass('board__square_color_red')).toEqual(true);
	});
});
