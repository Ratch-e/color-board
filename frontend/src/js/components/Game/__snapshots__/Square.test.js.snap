// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Квадрат на поле рендерится корректно 1`] = `1`;
